include/veriblock/blockchain/pop/pop_state_machine.hpp

```diff
@@ -26,19 +26,34 @@ struct PopStateMachine {
 
   PopStateMachine(ProtectedTree& ed,
                   ProtectingBlockTree& ing,
+                  PopStorage& storage,
                   height_t startHeight = 0)
-      : ed_(ed), ing_(ing), startHeight_(startHeight) {}
+      : ed_(ed), ing_(ing), storage_(storage), startHeight_(startHeight) {}
 
   bool applyBlock(index_t& index, ValidationState& state) {
     std::vector<CommandPtr> executed;
-    for (auto& cg : index.commands) {
+    for (const auto& pid : index.payloadIds) {
+      payloads_t payloads;
+      bool ret = storage_.loadPayloads(pid, payloads);
+      if (!ret) {
+        //TODO: error message
+        continue;
```

Comment:
throw here. something like 
```
struct StateCorruptionException: public std::domain_error {
  using base = std::domain_error;
  using base::base; // to inherit constructors
};
```

Reply:
even better if loadPayloads will not return true/false on each payloads, but rather work in batches, and if loading is not possible, then it would throw StateCorruptionException:
```
applyBlock(...) {
  std::vector<CommandGroup> groups = storage_.loadCommands(index.payloadIds);
  // ... normal flow
  
}
```

Reply:
Commands are constructed with tree reference. Should we pass payloadsToCommands lambda to storage_.loadCommands()?